# Introduction

This workshop provides a complete, start-to-finish guide to designing, analyzing, and interpreting bulk RNA-seq experiments, using web-based tools with no coding required. Participants will learn how to plan experiments, process sequencing data, and run analysis pipelines entirely through accessible online platforms. Emphasis is placed on quality control and generating accurate differential expression results. By the end of the workshop, attendees will be equipped to make informed decisions and interpret RNA-seq data with confidence.


**Course Objectives**

- Understand the key components and considerations for designing a robust bulk RNA-seq experiment.

- Gain familiarity with laxy.io for processing RNA-seq data, from sequencing to alignment and transcript counting.

- Perform quality control checks to ensure the accuracy and reliability of RNA-seq data.

- Generate and interpret differentially expressed genes (DEGs) using Degust.

- Develop the ability to critically evaluate and adjust RNA-seq experiments based on quality metrics and results.

## Scope

RNA-seq, as a technique, can be a broad topic. 

In this workshop, we will specifically cover using bulk **short read** 'next-generation' sequencing data for **differential expression analysis**, 
with a focus on using **web-based tools** and understanding **how design experiments** and **interpret quality metrics**.

The goal is that for many typical experimental designs, particpants will learn to more confidently approach their own data in order to process it 
from raw reads to a differential expression analysis.

In this workshop, we explicitly ***don't*** cover:

- Runnning pipelines and analysing data using the commandline, R, Python etc.
- How to do *De novo* assembly of transcriptomes
- Analysis of microRNAs (miRNAs), other small RNAs
- Analysis of long read technologies
- Single cell and spatial RNA-seq analysis
- How to discover novel transcripts / splice variants
- How to perform variant calling (SNPs) from RNA-seq data
