# Planning an RNAseq Experiment

> Useful links: https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0881-8#Sec2
>
> https://www.annualreviews.org/content/journals/10.1146/annurev-biodatasci-072018-021255

Broadly speaking, RNAseq is a next generation sequencing technique for profiling all or selected target RNA molecules in a given biological system. It typically involves isolating RNA, converting to cDNA, then ligating adapter sequences to the RNA before constructing a library that can be used for sequencing. A diverse ecosystem of protocols and technology exist that can be used to generate RNAseq data, which can be used in a wide variety of applications. 

```{r, echo=FALSE, out.width="100%", fig.cap="Overview of the steps in an RNAseq experiment. At each of these steps, there are choices that are made that can influence the final output of the experiment."}
knitr::include_graphics("images/experimental_design/overview_rnaseq.png")
```

Image source: [RNA Sequencing Data: Hitchhiker's Guide to Expression Analysis, 2019](https://www.annualreviews.org/content/journals/10.1146/annurev-biodatasci-072018-021255)


The first step of planning an RNAseq experiment is asking whether RNAseq will answer the intended research question and whether it should be performed. Sequencing experiments are not cheap and a lot of time and money can be saved by ensuring that the sequencing experiment performed is suited to answering the research question at hand. There is a huge diversity in what RNA-seq can achieve due to the number of different protocols that have been developed and published. Therefore having a clear experimental goal will help in selecting an appropriate protocol that will best answer the research question as well as a good experimental design that will allow you to get the statistical power to answer that question.

There are some main aspects to planning an RNAseq experiment:

1. What is the research goal?
2. What sequencing technology will be used - can the selected protocol answer the research question
3. What is the experimental design - will the number of collected samples have enough statistical power to answer the research question? Are there confounding factors that might obscure the effect of biology that we want to study?
4. What is the budget for the proposed experiment?
5. What sort of analysis can be performed with the data that has been generated from the experiment?

Having a clear research goal guides subsequent choices in terms of the type of RNAseq performed and the number of samples. The first, second and third aspects are limited and linked by cost. The ideal protocol for answering your research question might be prohibitively expensive - so you may opt for a different protocol. The ideal experimental design may require more samples than you can either afford to collect or are able to collect due to multitude of factors including ethical approval or sample rarity. It is very common for RNAseq experiments to be limited in the number of biological replicates collected due to the cost. 

```{r, out.width="100%", echo=FALSE, fig.cap="While the basic principles of experimental design are always strongly encouraged, the principle of replication unfortunately is often discarded in the face of sequencing costs"}
knitr::include_graphics("images/experimental_design/expectations_reality.jpg")
```

There is no single RNAseq protocol that is optimised for answering every research question - instead they may have different advantages and disadvantages that can make one protocol better suited for a particular task over another.

```{r, echo=FALSE, out.width="90%", eval=FALSE}
knitr::include_graphics("https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41368-021-00146-0/MediaObjects/41368_2021_146_Fig1_HTML.png?as=webp")
#Image source: [From bulk, single-cell to spatial RNA sequencing (2021)](https://www.nature.com/articles/s41368-021-00146-0)
```

There are many choices to be made in designing an experiment and it is easy to feel overwhelmed by these choices. Consulting with relevant experts such as sequencing providers and bioinformaticians prior to carrying out the experiment can aid in this process. It is also allows you to anticipate potential complexities that may arise in the analysis of the data and mitigate them. It might take several iterations and consultations to settle on a design that will achieve the most of your research outcomes. It's also important to have an idea of how the generated data can then be analysed - there is not much point generating data if it cannot be used.

RNAseq data can be used for a variety of purposes. The most common one is quantative analysis of gene expression changes, though isoform level differential analysis can also be performed. RNAseq can also simultaneously be used for discovery, such as detection of novel transcripts, alternate splicing, exon skipping, intron retention or fusion genes. In organisms without a reference genome, RNAseq data can be used for de novo transcriptome assembly. 

```{r, fig.cap="", echo=FALSE, out.width="100%", fig.show="hold", fig.align="center"}
knitr::include_graphics(c("https://www.azenta.com/sites/default/files/web-media-library/blog/quick-start-guide-rna-seq-data-analysis/fig_5_deseq1.png", "https://www.azenta.com/sites/default/files/web-media-library/blog/quick-start-guide-rna-seq-data-analysis/fig_6_deseq2.png", "https://www.azenta.com/sites/default/files/web-media-library/blog/quick-start-guide-rna-seq-data-analysis/fig_7_deseq3.png"))
```

Image source: [A Quick Start Guide to RNA-Seq Data Analysis](https://www.azenta.com/blog/quick-start-guide-rna-seq-data-analysis)

The most common type of RNAseq experiment is a short read bulk experiment for the purpose of identifying differentially expressed genes in a given organism. This workshop has been designed with this understanding that this is the type of analysis that most researchers intend to perform but it is not the only application of RNAseq. This chapter will give an overview of the types of RNAseq protocols available to illustrate the field but the rest of the workshop will focus on short read bulk RNAseq.

## Before you begin: questions to considerâ€¦

The questions below are a few that might help in designing an experiment:

- do you want to assay changes in gene expression? 
- Do you want to identify novel transcripts?
- do you intend to study transcriptomic changes in individual cells or the tissue level? (single versus bulk RNAseq)
- do you want to characterise isoforms/study alternative splicing? (short versus long read sequencing)
- will you pull down transcripts with a poly A tail (poly-A selection) or deplete ribosomal genes (ribo depletion)? Does your organism produce mRNA with a polyA tail?
- single end or paired end sequencing?
- does your organism of interest have an annotated reference genome or will you need to assemble a transcriptome?
- are you going to align against the genome or transcriptome?
- how many different conditions are you testing? How many replicates do you have for each condition? Do you have a control condition?
- how complex is the experiment? Will you be able to collect all the samples you need in one experiment or are you going to have to do multiple batches?
- how inbred is your organism? If the organism of interest is not inbred, what level of genetic variation exists between your individuals?
- what is your budget? How many samples can you afford to sequence?  

## RNAseq Sequencing Protocols

RNAseq protocols can be described in a number of different ways. They can be classified as to the length of the reads produced by the technology, e.g either short or long reads. They can categorised by RNA captured for sequencing, e.g all available genes or if only a subset of target genes are used. More recently, with the development of single cell technology, they can now also be classified as the type of biological material they work with, ie is it at the single cell level or at a bulk tissue level. 

These descriptions of RNAseq are not mutually exclusive and are in fact used in combination with each other. 
<style type="text/css">
.tg  {border-collapse:collapse;border-spacing:0;}
.tg td{border-color:black;border-style:solid;border-width:1px;font-family:Arial, sans-serif;font-size:14px;
  overflow:hidden;padding:10px 5px;word-break:normal;}
.tg th{border-color:black;border-style:solid;border-width:1px;font-family:Arial, sans-serif;font-size:14px;
  font-weight:normal;overflow:hidden;padding:10px 5px;word-break:normal;}
.tg .tg-f8tv{border-color:inherit;font-style:italic;text-align:left;vertical-align:top}
.tg .tg-0pky{border-color:inherit;text-align:left;vertical-align:top}
.tg .tg-8zwo{font-style:italic;text-align:left;vertical-align:top}
.tg .tg-0lax{text-align:left;vertical-align:top}
</style>
<table class="tg"><thead>
  <tr>
    <th class="tg-f8tv">Biological material<br></th>
    <th class="tg-0pky">Entire tissue: Bulk RNAseq</th>
    <th class="tg-0pky">Individual cells: Single cell RNAseq</th>
    <th class="tg-0pky"></th>
  </tr></thead>
<tbody>
  <tr>
    <td class="tg-8zwo">RNA tissue source<br></td>
    <td class="tg-0lax">Fresh</td>
    <td class="tg-0lax">Frozen</td>
    <td class="tg-0lax">FFPE</td>
  </tr>
  <tr>
    <td class="tg-f8tv">RNA species sequenced</td>
    <td class="tg-0pky">All RNA species captured: total RNAseq</td>
    <td class="tg-0pky">mRNA only: mRNAseq</td>
    <td class="tg-0pky">short RNA: small RNAseq</td>
  </tr>
  <tr>
    <td class="tg-f8tv">RNA isolation method</td>
    <td class="tg-0pky">Ribosomal depletion</td>
    <td class="tg-0pky">PolyA pulldown</td>
    <td class="tg-0pky">Size selection</td>
  </tr>
  <tr>
    <td class="tg-f8tv">Read length</td>
    <td class="tg-0pky">50-300 bases: short read sequencing</td>
    <td class="tg-0pky">10-100kb: long read sequencing</td>
    <td class="tg-0pky">100-300kb: ultra long read sequencing</td>
  </tr>
  <tr>
    <td class="tg-f8tv">Library Strandedness</td>
    <td class="tg-0pky">Original strand information not preserved: nonstranded library</td>
    <td class="tg-0pky">Original strand information preserved: Stranded library</td>
    <td class="tg-0pky"></td>
  </tr>
  <tr>
    <td class="tg-f8tv">Which end of the fragment is sequencing<br></td>
    <td class="tg-0pky">Just 1 end: Single end sequencing</td>
    <td class="tg-0pky">Both ends: Paired end sequencing</td>
    <td class="tg-0pky"></td>
  </tr>
  <tr>
    <td class="tg-f8tv">Commercial platforms</td>
    <td class="tg-0pky">Illumina, BGI, PacBio, Nanopore, 10x, Nanostring, etc</td>
    <td class="tg-0pky"></td>
    <td class="tg-0pky"></td>
  </tr>
</tbody></table>


### RNA Isolation Methods

```{r, echo=FALSE, out.width="100%", fig.cap="Different RNAseq isolation methods"}
knitr::include_graphics("https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4863231/bin/nihms768779f1.jpg")
```

Image source: [RNA Sequencing and Analysis (2015)](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4863231/)

After RNA is initially extracted, the RNA molecules are a mix of ribosomal RNA (rRNA), non-coding RNAs as well as messenger RNA (mRNA). The majority of RNA molecules are rRNA which are typically not of interest - a choice is then made on which RNA species are to be sequenced.

```{r, echo=FALSE, out.width="50%", fig.cap="Estimate of RNA levels in a typical mammalian cells"}
knitr::include_graphics("images/experimental_design/rna_content.jpg")
```

Image source: [Non-coding RNA: what is functional and what is junk? 2015](https://www.frontiersin.org/journals/genetics/articles/10.3389/fgene.2015.00002/full)

- Ribosomal depletion: this method removes rRNA, leaving the mRNA, precursor messenger RNA (pre-mRNA) and non-coding RNAs. This is used for whole transcriptome sequencing or total RNA sequencing: sequencing of all RNA molecules (rRNA exluded) and is useful studying non-coding RNAs in addition to mRNA. 

- PolyA pulldown: this targets only RNA with a polyA tail - enriching for mRNAs. This is used for mRNA sequencing and is used when only the coding RNAs are of interest.

- Size selection: this is typically used to enrich for small RNAs and therefore used for small RNAseq. These protocols are used when the goal is to study small RNAs.


### Short & Long Read Sequencing

```{r, echo=FALSE, fig.align="center", fig.cap="Alternative splicing and RNA technologies", out.width="95%"}
knitr::include_graphics("https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10043755/bin/fgene-14-997383-g002.jpg")
```

Image source: [Deshpande D, et al, RNA-seq data science: From raw data to effective interpretation. Frontier Genetics. 2023](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10043755/)

Short read sequencing typically refers to protocols that fragment RNA into smaller pieces before converting to cDNA and sequencing. These are typically highly accurate and cost effective methods for a variety of tasks. This type of data is typically used for gene expression studies.

Long read sequencing typically refers to technology that produce reads with length ranging from 10-100kb or 100-300kb (ultra long read). Long read sequencing can capture the full length of mRNA transcripts, which is useful when examining changes in isoforms or when performing de novo transcriptome assembly. Long read sequencing can have lower accuracy than short read methods but has been improving over the recent years.

### Bulk & Single RNA Sequencing

When working with bulk tissue samples, the expression profiles of individual cells is lost. A specific cell type might express high levels for a particular gene while another cell type might have low expression for that gene. With bulk sequencing, that signal is lost and instead an average profile of that gene is captured instead across the tissue. To retain information about the individual cells, single cell sequencing protocols have been developed and become quite popular. 

```{r, out.width="100%", echo=FALSE}
knitr::include_graphics("https://www.lexogen.com/wp-content/uploads/2024/01/Difference-between-scRNA-Seq-and-bulk-RNA-Seq.png")
```

Image source: [Classical Single-Cell RNA Sequencing: a comprehensive overview](https://www.lexogen.com/classical-single-cell-rna-sequencing-a-comprehensive-overview/)

There are trade-offs between bulk and single cell sequencing. For example, the dominant single cell sequencing technology, 10x, detects the most abundant genes and mostly captures polyadenylated mRNA at the 3' end or the 5' end. Lowly abundant genes are likely to be misssed and the 3'/5' captures means that there is limited capability for analyses that are focused on information in other parts of the transcript.

```{r, out.width="100%", echo=FALSE}
knitr::include_graphics("https://lizard.bio/hs-fs/hubfs/scs_blog1_table-1.png?width=2065&height=1260&name=scs_blog1_table-1.png")
```

Image source: [Single cell vs bulk sequencing](https://lizard.bio/knowledge-hub/single-cell-vs-bulk-sequencing)

Bulk RNAseq remains popular due to cheaper cost and lower analysis complexity. It is suited for tasks where global expression changes are of interest.

### Sequencing Depth & Coverage

```{r, echo=FALSE, fig.align="center", fig.cap="Sequencing Depth and Coverage", out.width="90%"}
knitr::include_graphics("https://3billion.io/img/news/depth-and-coverage_2023-08-25-051528_evsk.webp")
```

[Image source](https://3billion.io/blog/sequencing-depth-vs-coverage)

These two terms are usually used interchangeably when describing the number of reads aligning to a reference genome but they do refer to slightly different concepts.

The depth of an experiment is the number of times any given position in the genome is sequenced. For example if at nucleotide A, it has been sequenced 15 times, then nucleotide A has a depth of 15x. If nucleotide B has been sequenced 40 times, then it has 40x depth. This can be extended to the average depth across the genome - a sample sequenced to have an average 30 reads at every position in the genome can be referred to have depth of 30x. However, two sequenced samples of 30x depth might not be the same quality - the first might have a uniform distribution of reads across the genome while the second might have high depth in some locations while having gaps in other locations.

Sequencing coverage refers to how much of the known genome has been sequenced. Typically, the aim is to sequence the entire genome but some targeted protocols might only sequence a selected number of regions/genes. 

A common dilemna researchers are often faced with is a choice as to whether to sequence more deeply or sequence more samples. More biological replicates will provide better estimates of variance and more precise measures of gene expression than sequencing to a greater depth. It is generally advised to sequence more samples rather than sequence deeply if the option is there as early studies showed that more replicates provided more statistical power to identify differentially expressed genes over sequencing to a greater depth.

```{r, echo=FALSE, fig.align="center", fig.cap="Biological replicates provide more statistical power to detect differential genes than sequencing depth", out.width="90%"}
knitr::include_graphics("images/experimental_design/liu_y_bioinformatics_2014.jpg")
```

Image source: [Liu Y et al, RNA-seq differential expression studies: more sequence or more replication? Bioinformatics. 2014](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3904521/)

However, higher sequencing depth is necessary for detecting lowly expressed differentially expressed (DE) genes and for conducting isoform-level differential expression analysis. Genes with high amounts of transcripts are more likely to get sequenced than genes with low amounts of transcripts. So a gene that hasn't been detected in an experiment could either be lowly expressed or not expressed but hard to determine if the experiment has low coverage.


## RNAseq Uses

The most common use for RNAseq is to identify differentially expressed genes in a tissue of interest for a given organism. There are other applications RNA-seq that we will briefly discuss, this list is not an exhaustive one.

- Differential expression analysis: DGE, DTE, DTU
- Novel transcript/isoform discovery: identify new transcripts/isoforms that are not annotated in a reference
- De novo transcriptome assembly: if a reference genome for an organism is not available, the information from RNAseq reads can be assembled into contigs to form a transcriptome for protein coding genes
- Detect alternate splicing/differential isoform usage, detect changes in isoform abundance
- Gene fusion detection
- Epigenetic integration: combine transcriptomic and epigenetic information

### Differential Expression Analysis

The most common type of analysis performed with RNAseq data is differential gene expression (DGE) analysis. This type of analysis looks at the difference in abundance for genes between 2 or more conditions. The conditions should have a suitable control condition that provides a baseline measurement that another condition can be compared against.

```{r, echo=FALSE, fig.align="center", fig.cap ="With a bulk RNAseq experiment, we usually want to examine changes in transcript abundance in an organism. We do this by comparing the gene expression levels in an experimental condition against the gene expression levels in a control condition"}
knitr::include_graphics("images/experimental_design/fig_dge.png")
```


While DGE is the most common type of analysis performed on RNAseq data, it is not the only type of differential test that can be performed with RNAseq data. DGE analysis ignores what is happening at the isoform level and looks at global gene abundance changes only. Instead, an RNAseq dataset can be analysed to look at changes at the transcript level, to see if the change in conditions is also affecting multiple isoforms for a gene differently. 

```{r, echo=FALSE, fig.align="center", fig.cap ="In a DGE analysis, this gene might be ignored for having no change in expression when actually a switch is occuring amongst the isoforms that are being produced"}
knitr::include_graphics("images/experimental_design/fig_dif_splice_usage.png")
```

Lastly, we can look at whether transcript levels are changing between two conditions, regardless of the gene of origin. 
```{r, echo=FALSE, fig.align="center", fig.cap="The different types of differential expression analysis", out.width="100%"}
knitr::include_graphics("https://f1000research.s3.amazonaws.com/manuscripts/19594/95bdb2a3-7459-4d64-9569-ad69d8910d35_figure1.gif")
```

Image source: [Froussios K, et al, Relative Abundance of Transcripts (RATs): Identifying differential isoform abundance from RNA-seq. F1000Research. 2019](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10043755/)

If interested in transcript changes, the complexity of the analyses can be determined by the choice of short or long read sequencing. Short read sequencing is better suited for gene level analysis. Aligning to the transcriptome with short read data adds complexity to the analysis as many reads will be ambigiously mapped for transcripts. There are methods that have been developed for probabilistically assigning reads to transcripts.

```{r, echo=FALSE, out.width="100%"}
knitr::include_graphics("images/experimental_design/genome_transcriptome_alignment.png")
```

Image source: [RNA Sequencing Data: Hitchhiker's Guide to Expression Analysis, 2019](https://www.annualreviews.org/content/journals/10.1146/annurev-biodatasci-072018-021255)

Long read sequencing is ideal for isoform/transcript level analyses as 


### Novel Transcript/Isoform Discovery

Most RNAseq datasets are used for quantative analysis - the data is aligned to a pre-existing reference genome and pre-existing annotations. In well annotated organisms or tissues, there usually isn't much concern that some genes might be going undetected and therefore ignored in the subsequent analysis. However, non-model organisms or tissues that have been less studied, the annotations can be incomplete or non-existent. In such cases, the RNAseq data can be leveraged to simultaneously provide information on the gene annotation as well the quantative changes in expression.

It is possible to leverage the sequences of the reads to assemble into transcripts (or if using long read sequencing, to sequence the full transcript and no assembly is then required). This can enable discovery of new genes/transcripts that are not logged in a reference database.

```{r, echo=FALSE, out.width="100%"}
knitr::include_graphics("https://www.pacb.com/wp-content/uploads/img_isoform_discovery-1.svg")
```

Image source: [RNA sequencing, PacBio](https://www.pacb.com/products-and-services/applications/rna-sequencing/)

### Transcriptome Assembly

When a reference genome is either unavailable or not desired, it is possible to take RNAseq reads and assemble them into a transcriptome of the assayed genes. The assembly can be done in 2 ways, there are reference based methods and de novo methods. Reference based methods use the reference of either the organism or a closely related species. 

De novo assembly methods are reference free - this is useful when studying non-model organisms as often they lack well annotated reference genomes. In such situations, RNAseq data has a dual purpose - the reference is built from the sequences of the reads and then the reads are counted against the transcriptome for differential analysis.  

Long read sequencing is better suited to building high quality transcriptomes than short read sequencing as the longer read length retains more information and context about the original RNA molecule. A commonly used analogy when describing assembly is to imagine the genome as a puzzle that needs to be put back together. The smaller the pieces, the more difficult the puzzle. A puzzle made of larger pieces is much easier to put back together and reconstruct the original RNA sequence. 

```{r, echo=FALSE, out.width="100%"}
knitr::include_graphics("https://oup.silverchair-cdn.com/oup/backfile/Content_public/Journal/bib/23/2/10.1093_bib_bbab563/1/m_bbab563f3.jpeg?Expires=1729202148&Signature=FMhHgcKT3vS5SP1phAnOU3R9ayIzXS6DDCdTPOK7WQdpOPRSxFEqbIla6ISWV7pKDaINiPZBEdcpmjCn75qGFCIGTLasuKU~DD5Po69XlEH8Os~9UgmXbPMhGddL7cXxdw3skQ~dial4rI4A8mu388ZJlRPDbmr6EURBKAdm-lGpG~xxLJzN9hUmqB2FDNtOt6bY6TGvVRNOlGXTz24HveHp~feY4nhtpb9BpUEKBZgpChqMpQEw8mQAkmx-BcXBon8pynzZ7ud6cBjKMZ7W3T1Cip0OGSrJE2Bn~af~uY-Lv68PJ3319cnUelEs05Mnsr34ntJtnllPjVRUkDo2Bw__&Key-Pair-Id=APKAIE5G5CRDK6RD3PGA")
```

Image source: [A simple guide to de novo transcriptome assembly and annotation (2022)](https://academic.oup.com/bib/article/23/2/bbab563/6514404)


### Gene Fusion Detection

RNAseq can have clinical applications. One such application is the detection of fusion genes - these can arise due to chromosomal rearrangements combining the coding regions of two genes. These genes can produce aberrant proteins and lead to cancer development if the fused genes are oncogenes or tumor suppresor genes. Therefore, detection of fusion genes can be an important diagnostic tool in clinical settings as well as for cancer research.

```{r, echo=FALSE, out.width="100%"}
knitr::include_graphics("https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41598-017-07070-6/MediaObjects/41598_2017_7070_Fig2_HTML.jpg?as=webp")
```

Image source: [GFusion: an Effective Algorithm to Identify Fusion Genes from Cancer RNA-Seq Data, 2017](https://www.nature.com/articles/s41598-017-07070-6)

### Epigenetic Integration

> Refer to https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0881-8#Sec20
>
> What is it
> 
> Why would you want to do it
> 
> How would you do it


## Designing a Bulk RNAseq Experiment

> Knowing what you want to study will determine the protocol
>
> This should probably be the last section
>
> This section should have an example where there is a research question and then how you'd design an RNAseq experiment to answer that question
>
> Have an exercise here that builds on the previous sections to get participants to design their own experiment based on what they are interested in


The simplest experimental setup for an RNAseq experiment would be 2 conditions of interest: a control condition and an experimental condition. By comparing the gene expression levels in the experimental condition against the expression levels in the control condition, we can determine which genes are responding to the experimental condition. The experimental condition could be a drug treatment, knocking out a gene of interest, over expressing a gene of interest, a different type of diet, etc. The control condition is our baseline measurement - we compare the gene expression levels in our experimental condition to the gene expression levels in the control condition. 

> Maybe move these figures around


### Pooling Samples

> What is a pooled sample
>
> Why might we use a pooled sample (bulk RNAseq)
>
> If we've pooled a sample - does a pooled sample count as multiple biological samples or do we still need replicates 

Sometimes, it is difficut to get enough biological material from one individuals to generate a sample. In such a situation, a researcher might choose to collect material from multiple individuals and pool it together to generate a sample. This is called a pooled sample - we have combined the RNA from differents individuals together to get a single sample. This sometimes causes confusion as to whether this sample counts as multiple biological samples and if more replicates are needed. In most cases, this pooled sample counts as one biological replicate and more replicates are still required for your experiment.

Typically we do not have information about which individual an RNA molecule belongs to in the pooled sample. For each gene, we will have a single value - the combined amount from the different individuals that made up the pooled sample. 

If the RNA molecules are tagged or barcoded in such a way prior to pooling that we are then able to identify which individual the RNA molecule came from, then we can leverage that information 