# Experimental Design

## Introduction


The first step of planning an RNAseq experiment is asking whether RNAseq will answer the intended research question and whether it should be performed. Sequencing experiments are not cheap and a lot of time and money can be saved by ensuring that the sequencing experiment performed is suited to answering the research question at hand. There is a huge diversity in what RNA-seq can achieve due to the number of different protocols that have been  developed and published. Therefore having a clear experimental goal will help in selecting an appropriate protocol that will best answer the research question.

There are many choices to be made in designing an experiment and it is easy to feel overwhelmed by these choices. Consulting with relevant experts such as sequencing providers and bioinformaticians prior to performing the experiment can aid in this process. 

The most common use for RNAseq is to identify differentially expressed genes in a tissue of interest for a given organism. This workshop has been designed with this understanding that this is the type of analysis that most researchers intend to perform but it is not the only application of RNAseq. There are other applications RNA-seq that we will briefly discuss, this list is not an exhaustive one.

- Differential gene expression
- De novo transcriptome assembly: if a reference genome for an organism is not available, the information from RNAseq reads can be used
- Novel transcript/isoform discovery: identify new transcripts/isofrms that are not annotated in a reference
- Differential isoform usage: detect changes in isoform abundance
- Alternate splicing: 
- Gene fusion detection: Gene fusions can occur as a result of chromosomal re-arrangments, resulting in the exchange of coding or regulatory sequences between genes. This can be a cause of cancer and therefore identification of fusion genes has important implications for human health.
- Aberrant splicing event detection:
- Epigenetic integration: combine transcriptomic and epigenetic information


## RNAseq Applications


```{r, echo=FALSE, fig.align="center", fig.cap="Short read vs long read alignment"}
knitr::include_graphics("https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10043755/bin/fgene-14-997383-g002.jpg")
```

Image taken from: [Deshpande D, et al, RNA-seq data science: From raw data to effective interpretation. Frontier Genetics. 2023](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10043755/) - outlines long read and short read sequencing. - can highlight how you can use this to study/interrogate isoforms

## Before you begin: questions to consider …


Do you want to assay changes in gene expression? Or do you want to identify novel transcripts?

The questions below are a few that might help in selecting a suitable protocol:

- do you intend to study transcriptomic changes in individual cells or the tissue level? (single versus bulk RNAseq)
- do you want to characterise isoforms/study alternative splicing? (short versus long read sequencing)
- will you pull down transcripts with a poly A tail (poly-A selection) or deplete ribosomal genes (ribo depletion)
- single end or paired end sequencing?
- does your organism of interest have an annotated reference genome or will you need to assemble a transcriptome?


## Designing a Bulk RNAseq Experiment

> Knowing what you want to study will determine the protocol

It is important to have a clear idea of what the experimental goal is as this will guide the experimental design in terms of choice of sequencing protocol, technology and the number of replicates. 

The simplest experimental setup for an RNAseq experiment would be 2 conditions of interest: a control condition and an experimental condition. By comparing the gene expression levels in the experimental condition against the expression levels in the control condition, we can determine which genes are responding to the experimental condition. The experimental condition could be a drug treatment, knocking out a gene of interest, over expressing a gene of interest, a different type of diet, etc. The control condition is our baseline measurement - we compare the gene expression levels in our experimental condition to the gene expresison levels in the control condition. 


```{r, echo=FALSE, fig.align="center", fig.cap ="With a bulk RNAseq experiment, we usually want to examine changes in transcript abundance in an organism. We do this by comparing the gene expression levels in an experimental condition against the gene expression levels in a control condition"}
knitr::include_graphics("images/experimental_design/fig_experimental_design.png")
```

Fundamentals of research -> Factor , Confounding factor, Variable to measure, Controls, Batch, Techinical replicates and not biological, Pooling samples, 

### Experimental Factors



### Pooling Samples

## Sequencing methods

Sequencing Decision. Transcriptomics for Bulk RNAseq. Short reads. Coverage.

### Short Read Sequencing

### Long Read Sequencing

### Coverage


A common query is to what depth should a sequence experiment be performed. 

Researchers are often faced with a choice as to whether to sequence more deeply or sequence more samples. More biological replicates will provide better estimates of variance and more precise measures of gene expression than sequencing to a greater depth. It is generally advised to sequence more samples rather than sequence deeply if the option is there. 

```{r, echo=FALSE, fig.align="center", fig.cap="Biological replicates provide more statistical power to detect differential genes than sequencing depth", out.width="350px", out.height="300px"}
knitr::include_graphics("images/experimental_design/liu_y_bioinformatics_2014.jpg")
```

Image taken from: [Liu Y et al, RNA-seq differential expression studies: more sequence or more replication? Bioinformatics. 2014](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3904521/)

However, higher sequencing depth is necessary for detecting lowly expressed differentially expressed (DE) genes and for conducting isoform-level differential expression analysis.

## Batches and replicates (technical/biological)

There are two types of replicates in an experiment:

1.  Technical replicates are repeated measurements taken from the same biological source to accurately **assess technical variation in an experiment**.

```{r, echo=FALSE, out.width="430px", out.height="300px", fig.align="center", fig.cap ="Technical replicates: A single mouse sample, three replicates per condiion.", fig.height= 0.1, fig.width=0.8}
knitr::include_graphics("images/experimental_design/technical_reps.png")
```

For example, consider a simple RNA-Seq experiment comparing two conditions: normal and diseased. In this scenario, a single normal mouse provides three measurements—mouseX_normal_1, mouseX_normal_2, and mouseX_normal_3. Similarly, a single diseased mouse provides three measurements—mouseY_diseased_1, mouseY_diseased_2, and mouseY_diseased_3. This results in the following samplesheet:

| sample | condition | mouse | technical replicates |
|:----------:|:---------:|:----------:|:---------:|
|  mouseX_normal_1  |  normal | X  |  1 |
|  mouseX_normal_2  |  normal | X  |  2 |
|  mouseX_normal_3  |  normal | X  |  3 |
|  mouseY_diseased_1  |  diseased  | Y |  1 |
|  mouseY_diseased_2  |  diseased | Y  |  2 |
|  mouseY_diseased_3  |  diseased | Y  |  3 |

The samples mouseX_normal_1, mouseX_normal_2, and mouseX_normal_3 serve as technical replicates for the normal condition, while mouseY_diseased_1, mouseY_diseased_2, and mouseY_diseased_3 are technical replicates for the diseased condition.

2.  Biological replicates involve using different biological sources to **capture the biological variation between samples**.


```{r, echo=FALSE, out.width="500px", out.height= 400px", fig.align="center", fig.cap ="Biological and Technical replicates: Three mice sample (biological reps), three replicates (technical) per condition."}
knitr::include_graphics("images/experimental_design/biological_reps.png")
```

In the previous RNA-Seq experiment, if we increase the number of mice to three for each healthy and diseased state, we obtain the following samplesheet:

| sample | condition | mouse (biological replicates) | technical replicates |
|:----------:|:---------:|:----------:|:---------:|
|  mouseX1_normal_1  |  normal   | X1  |  1    |
|  mouseX1_normal_2  |  normal   | X1  |  2    |
|  mouseX1_normal_3  |  normal   | X1  |  3    |
|  mouseX2_normal_1  |  normal   | X2  |  1    |
|  mouseX2_normal_2  |  normal   | X2  |  2    |
|  mouseX2_normal_3  |  normal   | X2  |  3    |
|  mouseX3_normal_1  |  normal   | X3  |  1    |
|  mouseX3_normal_2  |  normal   | X3  |  2    |
|  mouseX3_normal_3  |  normal   | X3  |  3    |
| mouseY1_diseased_1 | diseased  | Y1  |  1    |
| mouseY1_diseased_2 | diseased  | Y1  |  2    |
| mouseY1_diseased_3 | diseased  | Y1  |  3    |
| mouseY2_diseased_1 | diseased  | Y2  |  1    |
| mouseY2_diseased_2 | diseased  | Y2  |  2    |
| mouseY2_diseased_3 | diseased  | Y2  |  3    |
| mouseY3_diseased_1 | diseased  | Y3  |  1    |
| mouseY3_diseased_2 | diseased  | Y3  |  2    |
| mouseY3_diseased_3 | diseased  | Y3  |  3    |

In this experiment, X1, X2, and X3 serve as biological replicates for the normal condition, while Y1, Y2, and Y3 serve as biological replicates for the diseased condition. For each biological replicate, we have three technical replicates for both conditions.

In the era of microarrays, technical replicates were essential. However, with the advent of current RNA-Seq technologies, technical variation is minimal compared to biological variation, making technical replicates less necessary.

Conversely, biological replicates are indispensable for differential expression analysis. Increasing the number of biological replicates enhances the accuracy of biological variation estimates and the precision of mean expression level estimates. This, in turn, leads to better data modeling and the identification of more differentially expressed genes.


## Confounding factors

A confounded RNA-Seq experiment is one in which the distinct effects of two different sources of variation in the data cannot be separated.

For instance, sex is known to significantly influence gene expression. If all the control mice in an experiment were female and all the treatment mice were male, the treatment effect would be confounded by sex. In this scenario, it would be impossible to distinguish the treatment effect from the effect of sex. Consider another scenario, if all the control mice are from cage A and all the treatment mice for cage B, the treatment effects are confounded by the cage. The living conditions in the two cages may vary which is indistinguishable from the treatment effects.

```{r, echo=FALSE, fig.align="center", fig.cap ="Confounding effects in RNASeq experiments: Three female mice in the contol group and three male mice in the treatment group.", out.width="400px", out.height="300px"}
knitr::include_graphics("images/experimental_design/confounding.png")
```

To avoid confounding, it is important to ensure the sample source/animals in each condition are all the same sex, age, litter, and batch, if possible. If this is not entirely possible, then make sure to split the animals equally between different conditions. It's best to discuss he experimental design with a bioinformatcian before conducting the experiment.

As shown in Liu, Zhou, and White (2013)

* gene-level differential expression
* Isoform-level differential expression
* Other types of RNA analyses (intron retention, small RNA-Seq, etc.)

Batch, Techinical replicates and not biological, Pooling samples, Randomization, Noise. Clonal (cell-lines, bacteria, inbred animalls) or not clonal organisms, Does my tissue has different cell types -> Cell sorting or single cell
