# Pipeline Overview

## High-level

An RNA-seq pipeline needs to identify which RNA transcript a (short) read has originated from to quantify transcription of genes.

Once we know the origin of each read, we can use this to estimate the abundance of each transcript.

The minimal inputs to an RNA-seq processing pipeline are:

- a set of **raw reads** (typically in gzipped FASTQ format)
- a **reference genome sequence** (FASTA, gzipped) and **gene annotations** (GTF or GFF3, gzipped)

At the high-level, a pipeline aims to:

- align or 'map' the read to the genome (or transcriptome)
  - _(mRNA transcript, exon, intron, or somewhere else on the genome ?)_
- count reads associated with features / quantify abundance

Counts of the number of reads associated with each feature (_gene_) are used to 
estimate transcript relative abundance through differential expression analysis.

## A simple pipeline

<a href="images/pipeline/simple_pipeline2.svg" target="_blank">
![Simple RNA-seq pipeline overview](images/pipeline/simple_pipeline.svg){width="100%"}
</a>

## A more complex pipeline

[nf-core/rnaseq](https://nf-co.re/rnaseq)

<a href="images/pipeline/nf-core_rnaseq.png" target="_blank">
![nf-core/rnaseq metromap](images/pipeline/nf-core_rnaseq.png){width="100%"}
</a>

### Pipeline activity

**Challenge:** Create your own fantasy RNA-seq processing pipeline

_list tools, challenge is to compose them in a sensible order_
