## An example dataset

For the following examples, we will use the results of an `nf-core/rnaseq` run on a published dataset from [Shanle et al, 2013](https://doi.org/10.1210/me.2013-1164). ([SRP062287](https://www.ncbi.nlm.nih.gov/Traces/study/?acc=SRP062287)).
This is an engineered triple-negative breast cancer cell line, treated with various combinations of estrogen, doxycycline, and controls.

<a href="https://doi.org/10.1210/me.2013-1164" target="_blank">
![Shanle et al, 2013 headline](images/Shanle_et_al_2013.jpg){width="100%"}
</a>


We've already run this dataset through the nf-core/rnaseq pipeline - here is the output on [laxy.io](https://laxy.io/#/job/3pLfQoLEuWeAnWh4H3Vvbv/?access_token=e0124ee9-c8ad-4164-b59f-ca2ae0ce4d2a).

## Raw read QC

Let's begin by looking at some metrics about the raw FASTQ reads. The most popular tool for this is FastQC, 
which generates reports on read quality, length, and content (among other things).

_guided tour through a FastQC report - pre and post trim. Trimming discussion_

_quality score discussion_

_while `FastQC` is the most popular tool, others exist - `fastp` also provides similar QC reports_


## Mapping


## MultiQC interpretation

[MultiQC](https://multiqc.info/) is a tool that help aggregate and visualize sequencing 
quality metrics from many tools into a single report. We will examine the report generated by
the `nf-core/rnaseq` pipeline using MultiQC, focusing on the parts that are most useful for understanding
your dataset.

> Here's a link to the report [on laxy.io](https://api.laxy.io/api/v1/job/3lQ6tw48pcFfOKyg1vTJry/files/output/results/multiqc/star_salmon/multiqc_report.html){target="_blank"} (and an alternative backup version [here](files/multiqc/SRP062287/multiqc_report.html){target="_blank"}).

We can use these QC metrics to identify:

- quality poor samples
- potential genomic DNA contamination
- ribosomal RNA contamination
- possible PCR derived duplication
- over-sequencing / under-sequencing
- reference genome mixups
- library prep anomalies, barcode mixups

By the time you see this QC, often there is very little that can be done to 'fix' a dataset or poor experimental design.
However, these quality issues do not nessecarily make a dataset unusable. It's useful to know what you are working with.
This might guide decisions around future experiments or resequencing, or occasionally provide justification for excluding some samples.


### MultiQC: General Statistics table

At the very top of the report is a table aggregating statistics across various tools. 
This is handy for seeing some key metrics all in one place.

There's a lot of columns here - you can use the "configure columns" button to show just a subset.

![MultiQC configure columns](images/multiqc/mutliqc_configure_columns.jpg)

Let's skip interpreting this table to begin with, since most of the information is repeated in plots below. 

### MultiQC: Duplication statistics

[See this section](files/multiqc/SRP062287/multiqc_report.html#picard)

`picard MarkDuplicates` identifies and quantifies 'duplicate' mapped reads - if two reads have the same 5' start position and the same orientation (strand) (prior to soft-clipping), one of these reads is considered a duplicate.

These dupicates can be of two origins:

1. Reads from generated from different individual transcript molecules
2. Reads generated by PCR of a single fragment, which are derived from only one transcript molecule

Duplicate reads in category (1) are 'good', in that they represent the real transcript abundance. The 'deeper' you sequence (the larger the library), the more duplicates you'll get.
Duplicate reads in category (2) aren't desired, since they inflate and bias the real transcript abundance. 

#### UMIs and dupRadar

This is the purpose of Universal Molecular Idenifiers (UMIs) - semi-random sequences incorporated into fragments as early as possible in library prep, prior to many rounds of PCR.
They allow duplicated generated by PCR to be identified and accounted for.

_[dupRadar](https://doi.org/10.1186/s12859-016-1276-2) can help assess if an experiment is suffering from excessive PCR duplication, vs. 'over-sequencing'_
