# Pipeline Overview

## High-level

An RNA-seq pipeline needs to identify where a (short) read has come from to quantify transcription of genes.

At the high-level, this usually involves:

- align or 'map' the read to the genome (or transcriptome)
  - _(mRNA transcript, or somewhere else on the genome ?)_
- count / quantify abundance
- differential expression analysis

## A simple pipeline

<a href="images/pipeline/simple_pipeline2.svg" target="_blank">
![Simple RNA-seq pipeline overview](images/pipeline/simple_pipeline.svg){width="100%"}
</a>

## A more complex pipeline

[nf-core/rnaseq](https://nf-co.re/rnaseq)

<a href="images/pipeline/nf-core_rnaseq.png" target="_blank">
![nf-core/rnaseq metromap](images/pipeline/nf-core_rnaseq.png){width="100%"}
</a>

### Pipeline activity

**Challenge:** Create your own fantasy RNA-seq processing pipeline

_list tools, challenge is to compose them in a sensible order_

## Raw read QC

What's inside a FASTQ file ?

```
@SRR5963435.1 FCC3UGWACXX:1:1101:1196:2111/1
AANGGGCGCAATGTGCGTTCAAAGATTCGATGATTCACGGAATTCTGCAATTCACATTACGTATCGCATTTCGCTGCGTTCTTCATCGATGCGAGAACCA
+
@C#4=BB=<FFHG>GAFHIGDGFHHGI:CD@>?BFGG>CEGIG>8BDFDGGEGEFFEHIIIGIIGGHBBBC>@CDBBDB??BDDED>9>AA?B@######
```

- FastQC: generates reports on read quality, length, and content

_guided tour through a FastQC report - pre and post trim. Trimming discussion_

_while `FastQC` is the most popular tool, others exist - `fastp` also provides similar QC reports_

## multiQC interpretation
## Mapping
## Proper references and annotated genome references
## multiQC interpretation
## Counting
## Feature counts

**Challenge:** Count reads for gene X

_IGV or IGV.js activity, manually count aligned reads in example, compare with featureCounts output_

## UMIs
## Show them where they my run it