## Alignment / mapping

![Simple RNA-seq pipeline overview: FastQC step](images/pipeline/simple_pipeline_alignment.svg){width="100%"}

> **TODO:** What aligners do, with a focus on STAR. Inputs (FASTQs), outputs (BAMs), multimappers, softclipping.

### Duplicate marking and sorting of alignment files

> **TODO:** Common tools are `samtools sort` and `picard MarkDuplicates`. Why ?

## MultiQC interpretation

[MultiQC](https://multiqc.info/) is a tool that help aggregate and visualize sequencing 
quality metrics from many tools into a single report. We will examine the report generated by
the `nf-core/rnaseq` pipeline using MultiQC, focusing on the parts that are most useful for understanding
your dataset.

> Here's a link to the report [on laxy.io](https://api.laxy.io/api/v1/job/3pLfQoLEuWeAnWh4H3Vvbv/files/output/results/multiqc/star_salmon/multiqc_report.html?access_token=e0124ee9-c8ad-4164-b59f-ca2ae0ce4d2a){target="_blank"} (and an alternative backup version [here](files/multiqc/SRP062287/multiqc_report.html){target="_blank"}).

We can use these QC metrics to identify:

- quality poor samples
- potential genomic DNA contamination
- ribosomal RNA contamination
- possible PCR derived duplication
- over-sequencing / under-sequencing
- reference genome mixups
- library prep anomalies, barcode mixups

By the time you see this QC, often there is very little that can be done to 'fix' a dataset or poor experimental design.
However, these quality issues do not nessecarily make a dataset unusable. It's useful to know what you are working with.
This might guide decisions around future experiments or resequencing, or occasionally provide justification for excluding some samples.


### MultiQC: General Statistics table

At the very top of the report is a table aggregating statistics across various tools. 
This is handy for seeing some key metrics all in one place.

There's a lot of columns here - you can use the "configure columns" button to show just a subset.

![MultiQC configure columns](images/multiqc/mutliqc_configure_columns.jpg){width="60%"}

Let's skip interpreting this table to begin with, since most of the information is repeated in plots below. 

### MultiQC: Duplication statistics

[See this section](files/multiqc/SRP062287/multiqc_report.html#picard)

`picard MarkDuplicates` identifies and quantifies 'duplicate' mapped reads - if two reads have the same 5' start position and the same orientation (strand) (prior to soft-clipping), one of these reads is considered a duplicate.
_(For paired end reads, both R1 and R2 5' start positions must match another pair)._

These duplicates can be of several origins:

1. Reads from generated from different individual transcript molecules
2. Reads generated by **PCR** of a single fragment, which are derived from only one transcript molecule
3. 'Optical duplicates' that originate from image analysis artefacts - where a single cluster on the flowcell is incorrectly interpreted as two (or more) clusters

Duplicate reads in category (1, biological) are generally 'good', in that they represent the real transcript abundance. The deeper you sequence (the larger the library), the more duplicates you'll get since the chance of two reads happening to have the same start position by chance increases.

Duplicate reads in category (2, PCR) usually aren't desired, since they may bias the real transcript abundance, inflating counts for fragments that happen to amplify more efficiently.

Duplicate reads in category (3, optical) seem to be rare in data on modern instruments, probably probably due to better image analysis and patterned flow cells. 

We can't _always_ know if a duplicate is biological (1), PCR-derived (2) or optical (3).

##### Optical duplicates

If the sequencing flowcell has tile coordinates (recorded in the FASTQ headers), it is possible to mark optical duplicates based on physical proximity. 
The instrument is likely to have dealt with this at the image analysis stage before we see the data.
_(Illumina instruments generally produce tile coordinates that `picard` and other tools can understand. MGI instruments ... ðŸ¤· ?)_

##### PCR duplicates

These duplicate reads are derived from PCR amplification of the library, usually in the case of low input libraries where 
more PCR cycles are used to increase the library size. Because some fragments can amplify more efficiently than others, and due
to the PCR 'jackpot' effect, we can end up with some fragments with high duplication due to PCR. 
This can bias the estimated abundance of transcripts.

There are two main solutions to avoid excessive PCR duplicates are:

- Get more starting material so you don't need to artificially increase the library size via many PCR cycles
- Use Universal Molecular Identifiers (UMIs) in the library construction to allow PCR duplicates to be identified.

#### UMIs and dupRadar

This is the purpose of Universal Molecular Identifiers (UMIs) - semi-random sequences incorporated into fragments as early as possible in library prep, prior to many rounds of PCR.
They allow duplicated generated by PCR to be identified and accounted for.

_[dupRadar](https://doi.org/10.1186/s12859-016-1276-2) can help assess if an experiment is suffering from excessive PCR duplication, vs. 'over-sequencing'_
