# Differential Expression

We will be looking at data from the project [SRP062287](https://www.ncbi.nlm.nih.gov/Traces/study/?acc=SRP062287) in the Sequence Read Archive (SRA). Read counts can be found in [GSE71960](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE71960) in Gene Expression Omnibus (GEO), but in this workshop we have produced our own read counts. The data is associated with [this publication](https://academic.oup.com/mend/article/27/10/1762/2615173).

Triple-negative breast cancer cell line MDA-MB-468 has been engineered to have an inducible ERÎ² estrogen receptor. This gene is expressed when treated with doxycycline (plusdox). Some samples are also be treated with estrogen E2. We expect interesting things to happen when both the estrogen receptor and the estrogen are present.

There are 12 samples in 4 conditions:

|sample     |condition    | replicate|
|:----------|:------------|---:|
|SRR2155413 |DMSO_nodox   |   1|
|SRR2155414 |DMSO_nodox   |   2|
|SRR2155415 |DMSO_nodox   |   3|
|SRR2155416 |E2_nodox     |   1|
|SRR2155417 |E2_nodox     |   2|
|SRR2155418 |E2_nodox     |   3|
|SRR2155419 |DMSO_plusdox |   1|
|SRR2155420 |DMSO_plusdox |   2|
|SRR2155421 |DMSO_plusdox |   3|
|SRR2155422 |E2_plusdox   |   1|
|SRR2155423 |E2_plusdox   |   2|
|SRR2155424 |E2_plusdox   |   3|

## A first look

These 12 read sets have been run through the nf-core/rnaseq pipeline using the [laxy.io](https://laxy.io) website.

&rarr; [laxy.io output](https://laxy.io/#/job/3lQ6tw48pcFfOKyg1vTJry/?access_token=485eaae2-46e7-4c15-8d83-6bd381615ae5)

<!--We will use the file `counts.star_featureCounts.tsv`, which counts reads unambiguously aligned to genes in a straightforward way. See later for a discussion of different ways of counting. -->

We will use `salmon.merged.gene_counts.biotypes.tsv`. You can find it in the Laxy output in the `output/results/star_salmon/` folder. Download this file and then upload it to [Degust](https://degust.erc.monash.edu/). We now need to configure Degust:

* In "Info columns" add "gene_id" and "gene_name".
* Using the "Add condition" button, add three conditions. Add samples to each condition as in the table above.
* Press "Save changes".

## Gene expression filter

At this point, Degust warns us there is no gene expression filter. This is important!

* Testing many genes increases the size of the FDR correction for multiple testing. Since there is no hope of detecting differential expression in lowly expressed genes, we prefer to remove these before testing. 
* The "Empirical Bayes" part of the statistical test goes wonky if genes with very low expression are included.

Click on "QC" in the top right and then "Library sizes". This shows the total count for each sample. This experiment has a variety of library sizes, with a typical library size around 25 million.

If few samples have a count of more than 5 to 10, we can safely discard them. Our rule of thumb is to require this in group size minus one samples. Here our group size is 3, so we will require at least 2 samples with a non-trivial count. So that we treat each sample in the same way, we will set a threshold on "Counts Per Million" (CPM) rather than the raw counts.

* Press "Configure" in the top right to return to the configuration screen.
* Set "Min gene CPM" to 0.5 and "in at least samples" to 2.
* Press "Save changes".

&rarr; [Degust page set up as described](https://degust.erc.monash.edu/degust/compare.html?code=d4453d6c1cbf8f7c0c876732a4bc0ba5#/)

**TODO: Discuss rules of thumb!**

## Normalisation

## analysis ie DEG

 ( should we go into more details on design matrices, contrast matrices, fitting linear models etc.?)

## QC of results
### MDA/PCA plots and running additional adjustments such as RUVseq or adding covariates in DGE


## Addendum: Which counts file to use?

What are you counting?

How should you count?

What about transcript level counts or differential exon usage?