% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{book}
\usepackage{amsmath,amssymb}
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math} % this also loads fontspec
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
\usepackage{lmodern}
\ifPDFTeX\else
  % xetex/luatex font selection
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\usepackage{longtable,booktabs,array}
\usepackage{calc} % for calculating minipage widths
% Correct order of tables after \paragraph or \subparagraph
\usepackage{etoolbox}
\makeatletter
\patchcmd\longtable{\par}{\if@noskipsec\mbox{}\fi\par}{}{}
\makeatother
% Allow footnotes in longtable head/foot
\IfFileExists{footnotehyper.sty}{\usepackage{footnotehyper}}{\usepackage{footnote}}
\makesavenoteenv{longtable}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{5}
\usepackage{booktabs}
\ifLuaTeX
  \usepackage{selnolig}  % disable illegal ligatures
\fi
\usepackage[]{natbib}
\bibliographystyle{plainnat}
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\urlstyle{same}
\hypersetup{
  pdftitle={RNAseq Workshop},
  pdfauthor={Monash Genomics and Bioinformatics Platform (MGBP)},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}

\title{RNAseq Workshop}
\author{Monash Genomics and Bioinformatics Platform (MGBP)}
\date{Compiled: May 06, 2024}

\begin{document}
\maketitle

{
\setcounter{tocdepth}{1}
\tableofcontents
}
\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{getting-started}{%
\chapter{Getting started}\label{getting-started}}

Instructors:

\hypertarget{summary}{%
\section{Summary}\label{summary}}

Important links:

\hypertarget{introduction}{%
\chapter{Introduction}\label{introduction}}

\hypertarget{experimental-design}{%
\chapter{Experimental design}\label{experimental-design}}

\hypertarget{sequencing-methods}{%
\section{Sequencing methods}\label{sequencing-methods}}

\hypertarget{cost-optimisation-approaches}{%
\section{Cost optimisation approaches}\label{cost-optimisation-approaches}}

\hypertarget{sequencing}{%
\section{Sequencing}\label{sequencing}}

\hypertarget{batches-and-replicates-technicalbiological}{%
\section{Batches and replicates (technical/biological)}\label{batches-and-replicates-technicalbiological}}

\hypertarget{library-preparation}{%
\chapter{Library Preparation}\label{library-preparation}}

\hypertarget{read-lengthtype-pairedsingle}{%
\section{Read length/type, paired/single}\label{read-lengthtype-pairedsingle}}

\hypertarget{enrichmentdepletion}{%
\section{Enrichment/depletion}\label{enrichmentdepletion}}

\hypertarget{cdna-synthesis}{%
\section{cDNA synthesis}\label{cdna-synthesis}}

\hypertarget{pipeline-overview}{%
\chapter{Pipeline Overview}\label{pipeline-overview}}

\hypertarget{sequencing-1}{%
\section{Sequencing}\label{sequencing-1}}

\hypertarget{fastqqc}{%
\section{(fastq)QC}\label{fastqqc}}

\hypertarget{multiqc-interpretation}{%
\section{multiQC interpretation}\label{multiqc-interpretation}}

\hypertarget{mapping}{%
\section{Mapping}\label{mapping}}

\hypertarget{proper-references-and-annotated-genome-references}{%
\section{Proper references and annotated genome references}\label{proper-references-and-annotated-genome-references}}

\hypertarget{multiqc-interpretation-1}{%
\section{multiQC interpretation}\label{multiqc-interpretation-1}}

\hypertarget{counting}{%
\section{Counting}\label{counting}}

\hypertarget{feature-counts}{%
\section{Feature counts}\label{feature-counts}}

\hypertarget{umis}{%
\section{UMIs}\label{umis}}

\hypertarget{show-them-where-they-my-run-it}{%
\section{Show them where they my run it}\label{show-them-where-they-my-run-it}}

\hypertarget{differential-expression}{%
\chapter{Differential Expression}\label{differential-expression}}

\hypertarget{normalisation}{%
\section{Normalisation}\label{normalisation}}

\hypertarget{analysis-ie-deg}{%
\section{analysis ie DEG}\label{analysis-ie-deg}}

( should we go into more details on design matrices, contrast matrices, fitting linear models etc.?)
\#\# QC of results
\#\#\# MDA/PCA plots and running additional adjustments such as RUVseq or adding covariates in DGE

\hypertarget{post-analysis}{%
\chapter{Post Analysis}\label{post-analysis}}

\hypertarget{discuss-additional-plots-and-variations}{%
\section{Discuss additional plots and variations}\label{discuss-additional-plots-and-variations}}

\hypertarget{discuss-network-analysis}{%
\section{Discuss Network analysis}\label{discuss-network-analysis}}

\hypertarget{discuss-multivariate-integration}{%
\section{Discuss Multivariate integration}\label{discuss-multivariate-integration}}

\hypertarget{discuss-single-cell}{%
\section{Discuss Single Cell}\label{discuss-single-cell}}

  \bibliography{book.bib,packages.bib}

\end{document}
